packages/data/src/types.ts

```diff
@@ -170,9 +170,17 @@ export type ConfigOf< S > = S extends StoreDescriptor< infer C > ? C : never;
 
 export type ActionCreatorsOf< Config extends AnyConfig > =
 	Config extends ReduxStoreConfig< any, infer ActionCreators, any >
-		? ActionCreators
+		? PromisifiedActionCreator< ActionCreators >
 		: never;
 
+// When dispatching an action creator, the return value is a promise.
+type PromisifiedActionCreator< ActionCreators extends MapOf< ActionCreator > > =
```

Comment:
The name should be `PromisifiedActionCreators` (plural) because it works with the entire map of actions, not just with an individual one.

`PromisifiedActionCreator` (singular) could be a nice helper that allows us to do the promisification in two parts:
```ts
type PromisifiedActionCreators< ActionCreators extends MapOf< ActionCreator > > =
  { [ Action in keyof ActionCreators ]: PromisifyActionCreator< ActionCreators[ Action ] > };

Reply:
Added